Fix reload callback session construction + remove duplication of code by refactoring the callback/promise handling code

oom- opened the pull request:
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/fastify/fastify/blob/master/CONTRIBUTING.md

By making a contribution to this project, I certify that:

* (a) The contribution was created in whole or in part by me and I
  have the right to submit it under the open source license
  indicated in the file; or

* (b) The contribution is based upon previous work that, to the best
  of my knowledge, is covered under an appropriate open source
  license and I have the right under that license to submit that
  work with modifications, whether created in whole or in part
  by me, under the same open source license (unless I am
  permitted to submit under a different license), as indicated
  in the file; or

* (c) The contribution was provided directly to me by some other
  person who certified (a), (b) or (c) and I have not modified
  it.

* (d) I understand and agree that this project and the contribution
  are public and that a record of the contribution (including all
  personal information I submit with it, including my sign-off) is
  maintained indefinitely and may be redistributed consistent with
  this project or the open source license(s) involved.
-->

Tests:
```
 PASS ​ test/memorystore.test.js 16 OK 63.406ms    
​ PASS ​ test/idGenerator.test.js 512 OK 140.972ms
​ PASS ​ test/fastifySession.checkOptions.test.js 12 OK 146.505ms
​ PASS ​ test/store.test.js 11 OK 189.105ms
​ PASS ​ test/base.test.js 62 OK 460.55ms
​ PASS ​ test/cookie.test.js 72 OK 508.147ms
​ PASS ​ test/session.test.js 123 OK 677.57ms
​ PASS ​ test/verifyPath.test.js 34 OK 116.583ms
​ PASS ​ test/expiration.test.js 5 OK 2s



  🌈 SUMMARY RESULTS 🌈  


Suites:   ​9 passed​, ​9 of 9 completed​
Asserts:  ​​​847 passed​, ​of 847​
​Time:​   ​4s
-------------------|---------|----------|---------|---------|-------------------
File               | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
-------------------|---------|----------|---------|---------|-------------------
All files          |     100 |      100 |     100 |     100 |
 cookie.js         |     100 |      100 |     100 |     100 |
 fastifySession.js |     100 |      100 |     100 |     100 |
 idGenerator.js    |     100 |      100 |     100 |     100 |
 session.js        |     100 |      100 |     100 |     100 |
 store.js          |     100 |      100 |     100 |     100 |
-------------------|---------|----------|---------|---------|-------------------
```

Benchmark (they do not run on my machine even with clean install):
```
╔══════════════╤═════════╤═════════╤═══════════╤═════════════════════════╗
║ Slower tests │ Samples │  Result │ Tolerance │ Difference with slowest ║
╟──────────────┼─────────┼─────────┼───────────┼─────────────────────────╢
║ redis        │       0 │ Errored │       N/A │ N/A                     ║
║ file         │       0 │ Errored │       N/A │ N/A                     ║
╟──────────────┼─────────┼─────────┼───────────┼─────────────────────────╢
║ Fastest test │ Samples │  Result │ Tolerance │ Difference with slowest ║
╟──────────────┼─────────┼─────────┼───────────┼─────────────────────────╢
║ memory       │       0 │ Errored │       N/A │ N/A                     ║
╚══════════════╧═════════╧═════════╧═══════════╧═════════════════════════╝

```

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added (no)
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

greenkeeper[bot] committed:
chore(package): update @types/node to version 13.7.6

Commit:
Merge pull request #95 from SerayaEryn/greenkeeper/@types/node-13.7.6

chore(package): update @types/node to version 13.7.6

greenkeeper[bot] committed:
chore(package): update ava to version 3.6.0

Commit:
Merge pull request #98 from SerayaEryn/greenkeeper/ava-3.6.0

chore(package): update ava to version 3.6.0

zekth committed:
Switch Store to map

onlyfortesting committed:
Support for Fastify v3

Commit:
Merge pull request #104 from onlyfortesting/master

Support for Fastify v3

Commit:
Merge pull request #101 from zekth/improve-default

Switch Store to map

Commit:
v4.0.0

greenkeeper[bot] committed:
chore(package): update @types/node to version 14.0.0

Commit:
Merge pull request #105 from SerayaEryn/greenkeeper/@types/node-14.0.0

Update @types/node to the latest version 🚀

Commit:
remove greenkeeper badge

Commit:
increase test coverage

Commit:
throw error on empty secret array

Commit:
export MemoryStore for mongo-connect

Commit:
v4.0.1

Commit:
Add Node 14 to CI

dependabot-preview[bot] committed:
Create Dependabot config file

Commit:
Merge pull request #108 from SerayaEryn/dependabot/add-v2-config-file

Create Dependabot config file

migramcastillo committed:
Fix deprecation alert for Fastify3.0.0

zekth committed:
fix: Typings for fastify 3

Commit:
Merge pull request #119 from zekth/fix_typings

fix: Typings for fastify 3

Commit:
Merge pull request #109 from migramcastillo/fix/deprecation-alert-fastifyv3

Fix deprecation alert for Fastify3.0.0

Commit:
v5.0.0

dependabot[bot] committed:
Bump typescript from 3.9.7 to 4.0.2

Bumps [typescript](https://github.com/Microsoft/TypeScript) from 3.9.7 to 4.0.2.
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v3.9.7...v4.0.2)

Signed-off-by: dependabot[bot] <support@github.com>

galvez committed:
feat: allow custom ID generator

Commit:
Merge pull request #124 from SerayaEryn/dependabot/npm_and_yarn/typescript-4.0.2

Bump typescript from 3.9.7 to 4.0.2

DRoet committed:
feat: replace deprecated request with got library

Commit:
Merge pull request #127 from DRoet/master

feat: replace deprecated request with got library

DRoet committed:
feat: add new decorate function for decrypting sessions

fixes #52

DRoet committed:
feat: add documentation about decryptSession

DRoet committed:
feat: use new parseCookie method from fastify-cookie

DRoet committed:
feat: expand example with parseCookie method

Commit:
Merge pull request #128 from DRoet/feat/decrypt-session

feat: add decorator to decrypt sessions outside requests

Commit:
v5.1.0

galvez committed:
add tests, small refactors

galvez committed:
fix tests

galvez committed:
fix conflicts

galvez committed:
fix

Commit:
Merge pull request #125 from galvez/dev

feat: allow custom ID generator

Commit:
v5.2.0

dependabot[bot] committed:
Bump fastify-plugin from 2.3.4 to 3.0.0

Bumps [fastify-plugin](https://github.com/fastify/fastify-plugin) from 2.3.4 to 3.0.0.
- [Release notes](https://github.com/fastify/fastify-plugin/releases)
- [Commits](https://github.com/fastify/fastify-plugin/compare/v2.3.4...v3.0.0)

Signed-off-by: dependabot[bot] <support@github.com>

DRoet committed:
fix: resolve fastify deprecation FST_LIGHTMYREQUEST_DEP01

Commit:
Merge pull request #130 from DRoet/master

fix: resolve fastify deprecation FST_LIGHTMYREQUEST_DEP01

Commit:
Merge pull request #129 from SerayaEryn/dependabot/npm_and_yarn/fastify-plugin-3.0.0

Bump fastify-plugin from 2.3.4 to 3.0.0

Commit:
update standard

Commit:
v5.2.1

mgcrea committed:
chore(types): add MemoryStore types

Commit:
Merge pull request #135 from mgcrea/fix-typings

chore(types): add MemoryStore types

zekth committed:
feat: migration

dependabot[bot] committed:
Bump fastify-cookie from 4.2.0 to 5.3.1 (#1)

dependabot[bot] committed:
Bump @types/node from 14.17.6 to 16.4.6 (#2)

zekth committed:
doc: add note (#3)

* doc: add note

* Update README.md

Co-authored-by: Manuel Spigolon <behemoth89@gmail.com>

Co-authored-by: Manuel Spigolon <behemoth89@gmail.com>

climba03003 committed:
Bumped v6.0.0

zekth committed:
doc: fix npm badge (#4)

vikasg603 committed:
Resolved FastifyDeprecation warning and added typescript documentation. (#6)

* Added typescript documentation

* Resolved issue related to FastifyDeprecation

* Removed deprecated FastifyPlugin import from types

* Update lib/fastifySession.js

Co-authored-by: KaKa <climba03003@gmail.com>

* Update README.md

Co-authored-by: Vincent LE GOFF <vince.legoff@gmail.com>

* Update using-store.ts

* Update types.d.ts

* Session Store ReadOnly

Co-authored-by: KaKa <climba03003@gmail.com>
Co-authored-by: Vincent LE GOFF <vince.legoff@gmail.com>

climba03003 committed:
Bumped v6.0.1

Fdawgs committed:
chore: use skeleton templates (#7)

* chore(gitignore): use skeleton template

* ci: use skeleton template; add automerge job

* docs(readme): use skeleton template badges

* Update README.md

Co-authored-by: Vincent LE GOFF <vince.legoff@gmail.com>

Co-authored-by: Vincent LE GOFF <vince.legoff@gmail.com>

dependabot[bot] committed:
Bump actions/setup-node from 2.3.2 to 2.4.0 (#8)

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 2.3.2 to 2.4.0.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v2.3.2...v2.4.0)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

zekth committed:
feat: enforce coverage 100% (#10)

dependabot[bot] committed:
Bump fastify/github-action-merge-dependabot from 2.2.0 to 2.3.0 (#12)

Bumps [fastify/github-action-merge-dependabot](https://github.com/fastify/github-action-merge-dependabot) from 2.2.0 to 2.3.0.
- [Release notes](https://github.com/fastify/github-action-merge-dependabot/releases)
- [Commits](https://github.com/fastify/github-action-merge-dependabot/compare/v2.2.0...v2.3.0)

---
updated-dependencies:
- dependency-name: fastify/github-action-merge-dependabot
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump fastify/github-action-merge-dependabot from 2.3.0 to 2.4.0 (#13)

Bumps [fastify/github-action-merge-dependabot](https://github.com/fastify/github-action-merge-dependabot) from 2.3.0 to 2.4.0.
- [Release notes](https://github.com/fastify/github-action-merge-dependabot/releases)
- [Commits](https://github.com/fastify/github-action-merge-dependabot/compare/v2.3.0...v2.4.0)

---
updated-dependencies:
- dependency-name: fastify/github-action-merge-dependabot
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

SimenB committed:
feat(types): export Store class (#15)

SimenB committed:
feat: set up session in `onRequest` rather than `preValidation` (#16)

climba03003 committed:
Bumped v6.1.0

SimenB committed:
fix(types): disallow unknown properties on session (#14)

* fix(types): disallow unknown properties on session

* add optional `id` to readme

climba03003 committed:
Bumped v6.1.1

AaronPorts committed:
Add rolling option (#20)

* Add rolling option

* Style fix

* Fix

* Add tests

* Add type test

climba03003 committed:
Bumped v6.2.0

zekth committed:
feat: switch to tsd (#22)

* feat: switch to tsd

* feat: missing assertions

* fix: review

* Update types/types.test-d.ts

Co-authored-by: Simen Bekkhus <sbekkhus91@gmail.com>

Co-authored-by: Simen Bekkhus <sbekkhus91@gmail.com>

ariesclark committed:
fix: idGenerator isn't in the options type. (#26)

* fix: idGenerator isn't in the options type.

* Add test for idGenerator

dependabot[bot] committed:
build(deps): bump actions/checkout from 2.3.4 to 2.3.5 (#28)

Bumps [actions/checkout](https://github.com/actions/checkout) from 2.3.4 to 2.3.5.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2.3.4...v2.3.5)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/setup-node from 2.4.0 to 2.4.1 (#25)

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 2.4.0 to 2.4.1.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v2.4.0...v2.4.1)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

lokshunhung committed:
fix: options both expires maxage (#30)

SimenB committed:
fix(types): type `get` & `set` correctly (#32)

* fix(types): type `get` & `set` correctly

* constructor as well

dependabot[bot] committed:
Bump fastify/github-action-merge-dependabot from 2.4.0 to 2.5.0 (#23)

Bumps [fastify/github-action-merge-dependabot](https://github.com/fastify/github-action-merge-dependabot) from 2.4.0 to 2.5.0.
- [Release notes](https://github.com/fastify/github-action-merge-dependabot/releases)
- [Commits](https://github.com/fastify/github-action-merge-dependabot/compare/v2.4.0...v2.5.0)

---
updated-dependencies:
- dependency-name: fastify/github-action-merge-dependabot
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump tsd from 0.17.0 to 0.18.0 (#27)

Bumps [tsd](https://github.com/SamVerschueren/tsd) from 0.17.0 to 0.18.0.
- [Release notes](https://github.com/SamVerschueren/tsd/releases)
- [Commits](https://github.com/SamVerschueren/tsd/compare/v0.17.0...v0.18.0)

---
updated-dependencies:
- dependency-name: tsd
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

climba03003 committed:
Bumped v6.2.1

dependabot[bot] committed:
build(deps): bump actions/checkout from 2.3.5 to 2.4.0 (#33)

Bumps [actions/checkout](https://github.com/actions/checkout) from 2.3.5 to 2.4.0.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2.3.5...v2.4.0)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

climba03003 committed:
fix: session (#36)

climba03003 committed:
Bumped v6.2.2

lokshunhung committed:
test: fix title (#38)

dependabot[bot] committed:
build(deps): bump fastify/github-action-merge-dependabot (#41)

Bumps [fastify/github-action-merge-dependabot](https://github.com/fastify/github-action-merge-dependabot) from 2.5.0 to 2.6.0.
- [Release notes](https://github.com/fastify/github-action-merge-dependabot/releases)
- [Commits](https://github.com/fastify/github-action-merge-dependabot/compare/v2.5.0...v2.6.0)

---
updated-dependencies:
- dependency-name: fastify/github-action-merge-dependabot
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump tsd from 0.18.0 to 0.19.0 (#42)

Bumps [tsd](https://github.com/SamVerschueren/tsd) from 0.18.0 to 0.19.0.
- [Release notes](https://github.com/SamVerschueren/tsd/releases)
- [Commits](https://github.com/SamVerschueren/tsd/compare/v0.18.0...v0.19.0)

---
updated-dependencies:
- dependency-name: tsd
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Fdawgs committed:
build(dependabot): ignore minor and patch github-actions updates (#43)

dependabot[bot] committed:
build(deps): bump fastify/github-action-merge-dependabot (#44)

Bumps [fastify/github-action-merge-dependabot](https://github.com/fastify/github-action-merge-dependabot) from 2.6.0 to 2.7.0.
- [Release notes](https://github.com/fastify/github-action-merge-dependabot/releases)
- [Commits](https://github.com/fastify/github-action-merge-dependabot/compare/v2.6.0...v2.7.0)

---
updated-dependencies:
- dependency-name: fastify/github-action-merge-dependabot
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump fastify/github-action-merge-dependabot (#45)

Bumps [fastify/github-action-merge-dependabot](https://github.com/fastify/github-action-merge-dependabot) from 2.7.0 to 2.7.1.
- [Release notes](https://github.com/fastify/github-action-merge-dependabot/releases)
- [Commits](https://github.com/fastify/github-action-merge-dependabot/compare/v2.7.0...v2.7.1)

---
updated-dependencies:
- dependency-name: fastify/github-action-merge-dependabot
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/node from 16.11.14 to 17.0.0 (#49)

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 16.11.14 to 17.0.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

sameer-coder committed:
Add Get/Set methods (#50)

* chore: added get/set methods

* chore: added type tests

climba03003 committed:
Bumped v6.3.0

zbo14 committed:
feat: session cookie opts (#53)

* feat: add cookieOpts to decryptSession

* chore: update readme documentation

* chore: add explanatory comment

mcollina committed:
Bumped v6.4.0

Signed-off-by: Matteo Collina <hello@matteocollina.com>

Fdawgs committed:
Update dependabot.yml (#57)

Fdawgs committed:
build(deps): bump fastify/github-action-merge-dependabot from 2.7.1 to 3 (#59)

dependabot[bot] committed:
build(deps): bump actions/setup-node from 2 to 3 (#60)

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 2 to 3.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

lokshunhung committed:
docs: typescript quirks (#40)

darkgl0w committed:
fix (types): fix callbacks types and ensure compatibility with `express-session` stores (#61)

* fix (types): fix callbacks types and ensure types compatibility with `express-session` stores

* test (types): fix callback tests

* fix (test/types): import session from plugin instead of types

* test (types): using a callback w/o args does not trigger an error

Legolaszstudio committed:
idGenerator request object (#62) (#64)

* Add request obj to idGenerator

* Update readme

* Fix null chain

* Make request param in regenerate optional

* Make request param optional

* Add customId test with rolling: false

* Add suggested rework for null checking

Co-authored-by: darkgl0w <31093081+darkgl0w@users.noreply.github.com>

* Add suggested changes

* Add type tests

* Remove optional chaining from test

Co-authored-by: darkgl0w <31093081+darkgl0w@users.noreply.github.com>

mcollina committed:
Bumped v6.5.0

Signed-off-by: Matteo Collina <hello@matteocollina.com>

ramonmulia committed:
major: drop node 10 and ava update (#65)

* chore: fix tests due major ava update

* fix: removing ci node10v due ava@4.1.0 compatibility

zekth committed:
feat: replace got by undici (#67)

* feat: replace got by undici

* fix: review

zekth committed:
fix: regex limitation (#68)

SimenB committed:
fix(types): error is optional, and session data always exists (#74)

* fix(types): error is optional, and session data always exists

* remove ifs

climba03003 committed:
Bumped v6.5.1

SimenB committed:
fix: remove request argument from Session#regenerate (#75)

Co-authored-by: KaKa <kaka@kakawebsitedemo.com>

climba03003 committed:
Bumped v7.0.0

SimenB committed:
feat: add `destroy`, `save` and `reload` to session instance (#77)

* feat: add `destroy`, `save` and `reload` to session instance

* tests

* somewhat simpler test

* Update types/types.test-d.ts

SimenB committed:
feat: Promise API (#78)

mcollina committed:
Bumped v8.0.0

Signed-off-by: Matteo Collina <hello@matteocollina.com>

dependabot[bot] committed:
build(deps-dev): bump tsd from 0.19.1 to 0.20.0 (#83)

Bumps [tsd](https://github.com/SamVerschueren/tsd) from 0.19.1 to 0.20.0.
- [Release notes](https://github.com/SamVerschueren/tsd/releases)
- [Commits](https://github.com/SamVerschueren/tsd/compare/v0.19.1...v0.20.0)

---
updated-dependencies:
- dependency-name: tsd
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

SimenB committed:
chore: add "benchmark" (#84)

* chore: add "benchmark"

* promise all

* actually 25

* no sinon here

SimenB committed:
feat: remove stale cookies (#79)

SimenB committed:
feat: only save session to storage when it changes (#80)

* feat: only save session to storage when it changes

* use safe-stable-stringify

* stringify is prod dep

climba03003 committed:
Bumped v8.1.0

dependabot[bot] committed:
build(deps-dev): bump undici from 4.16.0 to 5.0.0 (#82)

dependabot[bot] committed:
build(deps): bump actions/checkout from 2 to 3 (#81)

wilkmaia committed:
feat: add support for handling previously-signed cookies (#86)

* fix: Unsign cookie, if already signed by cookie plugin

* chore: add tests for new option

* chore: store default secret in variable on type tests

* chore: add type tests for new option

* fix: remote unnecessary test.plan

* feat: improve signed cookie handling

* chore: improve new test

* fix: fix support for node 12

Co-authored-by: m0dch3n <benoitthinnes@gmail.com>

dependabot[bot] committed:
build(deps-dev): bump standard from 16.0.4 to 17.0.0 (#87)

Bumps [standard](https://github.com/standard/standard) from 16.0.4 to 17.0.0.
- [Release notes](https://github.com/standard/standard/releases)
- [Changelog](https://github.com/standard/standard/blob/master/CHANGELOG.md)
- [Commits](https://github.com/standard/standard/compare/v16.0.4...v17.0.0)

---
updated-dependencies:
- dependency-name: standard
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mcollina committed:
Bumped v8.2.0

Signed-off-by: Matteo Collina <hello@matteocollina.com>

dependabot[bot] committed:
build(deps-dev): bump sinon from 13.0.2 to 14.0.0 (#88)

Bumps [sinon](https://github.com/sinonjs/sinon) from 13.0.2 to 14.0.0.
- [Release notes](https://github.com/sinonjs/sinon/releases)
- [Changelog](https://github.com/sinonjs/sinon/blob/main/docs/changelog.md)
- [Commits](https://github.com/sinonjs/sinon/compare/v13.0.2...v14.0.0)

---
updated-dependencies:
- dependency-name: sinon
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mcollina committed:
update to Fastify v4 (#89)

* update to Fastify v4

* drop node 12

* lower codecov to please windows

mcollina committed:
Bumped v9.0.0

ShogunPanda committed:
feat: Update benchmark. (#91)

Fdawgs committed:
ci: migrate to fastify reusable workflow (#90)

dependabot[bot] committed:
build(deps-dev): bump fastify-cookie from 5.6.1 to 5.7.0 (#92)

Bumps [fastify-cookie](https://github.com/fastify/fastify-cookie) from 5.6.1 to 5.7.0.
- [Release notes](https://github.com/fastify/fastify-cookie/releases)
- [Commits](https://github.com/fastify/fastify-cookie/commits)

---
updated-dependencies:
- dependency-name: fastify-cookie
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Fdawgs committed:
docs(readme): remove snyk badge (#93)

see https://github.com/fastify/fastify/issues/3883

Fdawgs committed:
chore(.gitignore): use updated skeleton template (#96)

dependabot[bot] committed:
build(deps-dev): bump @types/node from 17.0.45 to 18.0.0 (#97)

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 17.0.45 to 18.0.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump tsd from 0.20.0 to 0.21.0 (#98)

Bumps [tsd](https://github.com/SamVerschueren/tsd) from 0.20.0 to 0.21.0.
- [Release notes](https://github.com/SamVerschueren/tsd/releases)
- [Commits](https://github.com/SamVerschueren/tsd/compare/v0.20.0...v0.21.0)

---
updated-dependencies:
- dependency-name: tsd
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump tsd from 0.21.0 to 0.22.0 (#102)

Bumps [tsd](https://github.com/SamVerschueren/tsd) from 0.21.0 to 0.22.0.
- [Release notes](https://github.com/SamVerschueren/tsd/releases)
- [Commits](https://github.com/SamVerschueren/tsd/compare/v0.21.0...v0.22.0)

---
updated-dependencies:
- dependency-name: tsd
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Fdawgs committed:
style(types/types.test-d.ts): explicitly declare semicolon (#103)

Fdawgs committed:
refactor: use optional chaining (#104)

dependabot[bot] committed:
build(deps): bump fastify-plugin from 3.0.1 to 4.0.0 (#105)

Bumps [fastify-plugin](https://github.com/fastify/fastify-plugin) from 3.0.1 to 4.0.0.
- [Release notes](https://github.com/fastify/fastify-plugin/releases)
- [Commits](https://github.com/fastify/fastify-plugin/compare/v3.0.1...v4.0.0)

---
updated-dependencies:
- dependency-name: fastify-plugin
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Fdawgs committed:
docs: use `npm install` alias

See https://github.com/fastify/fastify/pull/3875

Uzlopak committed:
integrate metadata.js (#107)

Uzlopak committed:
separate lint from npm test script (#108)

activate lint and license-check in ci pipeline

Uzlopak committed:
refactor unit tests, replace ava with tap (#109)

Uzlopak committed:
move benchmark to corresponding folder, slim it down (#110)

* move benchmark to corresponding folder

* fix linting issue

* separate benchmark deps from project deps

* dont lint node_modules of benchmark

* remove one package.json

Uzlopak committed:
test: tap should only run test.js files (#115)

rclmenezes committed:
Clean up tests (#117)

* Use more constants in tests

* Test that maxAge also sends a cookie

* Clean up with invalid session test
- Make bad signature more obvious
- Remove expiration from existing session, which is a distraction

rclmenezes committed:
Add missing documentation for README.md (#116)

* Add missing documentation for README.md

* Update README.md

* Update README.md

Co-authored-by: Uzlopak <aras.abbasi@googlemail.com>

rclmenezes committed:
Simplify shouldSaveSession code a little bit (#119)

* Simplify shouldSaveSession code a little bit

* Even cleaner

rclmenezes committed:
Move `expires` from session to session.cookies (#120)

* Remove expires property from session
- Instead, put it in cookie

* Update docs

* Rm expires from addDataToSession

* Apply suggestions from code review

Co-authored-by: Uzlopak <aras.abbasi@googlemail.com>

Uzlopak committed:
chore: refactor store to es6 class (#122)

Uzlopak committed:
simplify callback typing (#124)

rclmenezes committed:
Add cookiePrefix as an option to allow for compatibility with express-session (#113)

* Add cookiePrefix as an option to allow for compatibility with express-session

* Add docs

* Fix lint

* improve solution

* Apply suggestions from code review

Co-authored-by: uzlopak <aras.abbasi@googlemail.com>

Uzlopak committed:
perf: improve idgenerator (about 25-30 % faster) (#123)

* add idgenerator

* remove uid-safe

* fix documentation

* use cached randomBytes for idGenerator

* fix linting issue

* simplify

* remove dot

Uzlopak committed:
remove addDataToSession (#127)

Uzlopak committed:
add autocannon example (#128)

* add autocannon example

* Update examples/memory.js

Co-authored-by: Rafael Gonzaga <rafael.nunu@hotmail.com>

Co-authored-by: Rafael Gonzaga <rafael.nunu@hotmail.com>

Uzlopak committed:
feat: use @fastify/cookie-signer-logic for signing and unsigning (#129)

* update fastify cookie to v8

* move all the functions into the session

* use Signer of @fastify/cookie

* remove anything related to unsignSignedCookie

* make it possible to have @fastify/cookie be loaded dynamically

* add the ability to set algorithm and signers

Uzlopak committed:
move signer require into ensureDefaults (#130)

Uzlopak committed:
fix unit test (#132)

* fix unit test

* make ensureDefaults immutable

* fix merge error

Uzlopak committed:
fix lgtm alert (#133)

rclmenezes committed:
Remove session id round 4 (#134)

* Rm sessionId from being stored in session

* Add a unit test to ensure we store the expected things

* Lint fix

* Readd session.sessionId assertion

* Do not shorten variable name

* improve unit test

* improve unit test

* fix lint issue

* make unit tests in session.test.js more verbose

Co-authored-by: Uzlopak <aras.abbasi@googlemail.com>

Uzlopak committed:
simplify getExpires, export getExpires from cookie (#135)

* simplify getExpires

* add exports for cookie

Uzlopak committed:
remove internal used session restore() function (#136)

Uzlopak committed:
Revert "chore: refactor store to es6 class (#122)" (#138)

This reverts commit 7dc9452b6a88e81d5de86ff0cfd3446ab4cc59f5.

Uzlopak committed:
remove sinon as dev dependency (#139)

* remove sinon as dev dependency

* fix lint issue

* remove unnecessary function

Uzlopak committed:
add redis example (#142)

Uzlopak committed:
add pre-commit (#145)

Uzlopak committed:
add unit test for maxAge (#144)

* add unit test for maxAge

* improve test

* Update test/session.test.js

Uzlopak committed:
create a separate unit test file for memorystore (#150)

* create a separate unit test file for memorystore

* fix test title

Uzlopak committed:
rename secret.test.js to fastifySession.checkOptions.test.js and refactor it (#151)

* rename secret.test.js to fastifySession.checkOptions.test.js

* improve unit tests for fastifySession.checkOptions.test.js

Uzlopak committed:
prevent encryptedSessionId from being stored in the sessionStore (#143)

* prevent encryptedSessionId from being stored in the sessionStore

* add one more condition

* improve test

* add TestStore, fix unit test

* reformat

* Update test/base.test.js

Co-authored-by: Simen Bekkhus <sbekkhus91@gmail.com>

* use includes instead of indexOf

* simplify TestStore

Co-authored-by: Simen Bekkhus <sbekkhus91@gmail.com>

Uzlopak committed:
use maxAge over expires (#152)

dependabot[bot] committed:
build(deps-dev): bump tsd from 0.22.0 to 0.23.0 (#153)

Bumps [tsd](https://github.com/SamVerschueren/tsd) from 0.22.0 to 0.23.0.
- [Release notes](https://github.com/SamVerschueren/tsd/releases)
- [Commits](https://github.com/SamVerschueren/tsd/compare/v0.22.0...v0.23.0)

---
updated-dependencies:
- dependency-name: tsd
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

rclmenezes committed:
Resave session cookie if saveUninitialized is false and rolling is true (#149)

* Clean up dates

* Clean up some code around cookiePrefix

* Implement rolling correctly

* Set request.session later, just in case

* Shorten variable names

* Check hasCookePrefix before string template

* use optimized getCookieSessionId

Co-authored-by: Uzlopak <aras.abbasi@googlemail.com>

Uzlopak committed:
rename session to fastifySession (#160)

Uzlopak committed:
Make it possible to call save() in decryptSession (#161)

* add sessionStore reference to session

* use direct reference to sessionStore

* multiline parameter of Session-
constructor

* add unit test

Uzlopak committed:
Implement originalMaxAge (#162)

* originalMaxAge initial

* add unit test for originalMaxAge

shrihari-prakash committed:
Reversed deprecation of decryptSession to support sharing sessions to websockets. (#164)

Uzlopak committed:
add typings for decryptSession (#165)

Uzlopak committed:
simplify cookie handling (#166)

Uzlopak committed:
chore: minor cosmetic change (#167)

Uzlopak committed:
ensure that secure false is not overwritten by null (#168)

Uzlopak committed:
security: verify path of cookie/session (#169)

* add tests

* patch path verification

* fix numbering

* add test

* dont spin up fastify for unit tests in verifyPath

climba03003 committed:
Bumped v10.0.0

dependabot[bot] committed:
build(deps-dev): bump tsd from 0.23.0 to 0.24.1 (#171)

Bumps [tsd](https://github.com/SamVerschueren/tsd) from 0.23.0 to 0.24.1.
- [Release notes](https://github.com/SamVerschueren/tsd/releases)
- [Commits](https://github.com/SamVerschueren/tsd/compare/v0.23.0...v0.24.1)

---
updated-dependencies:
- dependency-name: tsd
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Luchanso committed:
docs: update @fastify/cookie (#172)

HPieters committed:
fix rolling location in doc (#175)

TheWashiba committed:
fix: set domain when clearing cookie (#174)

* fix: set domain when clearing cookie

* fix: test case updated

* fix: updated tests

Eomm committed:
Bumped v10.0.1

climba03003 committed:
fix: align cookie implementation with express-session (#177)

* fix: align cookie implementation with express-session

* test: fix flaky test

climba03003 committed:
Bumped v10.0.2

Uzlopak committed:
nodenext compatibility (#178)

mcollina committed:
Bumped v10.1.0

Signed-off-by: Matteo Collina <hello@matteocollina.com>

dependabot[bot] committed:
build(deps-dev): bump tsd from 0.24.1 to 0.25.0 (#179)

Bumps [tsd](https://github.com/SamVerschueren/tsd) from 0.24.1 to 0.25.0.
- [Release notes](https://github.com/SamVerschueren/tsd/releases)
- [Commits](https://github.com/SamVerschueren/tsd/compare/v0.24.1...v0.25.0)

---
updated-dependencies:
- dependency-name: tsd
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

FabianFrank committed:
types: fix Session interface reference (#182)

mcollina committed:
Bumped v10.0.1

Signed-off-by: Matteo Collina <hello@matteocollina.com>

Fdawgs committed:
chore(.gitignore): add clinic (#183)

Fdawgs committed:
chore(.gitignore): add bun lockfile (#185)

dependabot[bot] committed:
build(deps-dev): bump connect-redis from 6.1.3 to 7.0.0 (#186)

Bumps [connect-redis](https://github.com/tj/connect-redis) from 6.1.3 to 7.0.0.
- [Release notes](https://github.com/tj/connect-redis/releases)
- [Commits](https://github.com/tj/connect-redis/compare/v6.1.3...v7.0.0)

---
updated-dependencies:
- dependency-name: connect-redis
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump tsd from 0.25.0 to 0.27.0 (#187)

Bumps [tsd](https://github.com/SamVerschueren/tsd) from 0.25.0 to 0.27.0.
- [Release notes](https://github.com/SamVerschueren/tsd/releases)
- [Commits](https://github.com/SamVerschueren/tsd/compare/v0.25.0...v0.27.0)

---
updated-dependencies:
- dependency-name: tsd
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump tsd from 0.27.0 to 0.28.0 (#188)

Bumps [tsd](https://github.com/SamVerschueren/tsd) from 0.27.0 to 0.28.0.
- [Release notes](https://github.com/SamVerschueren/tsd/releases)
- [Commits](https://github.com/SamVerschueren/tsd/compare/v0.27.0...v0.28.0)

---
updated-dependencies:
- dependency-name: tsd
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ojeytonwilliams committed:
fix: express session type compatibility (#190)

* fix: make types compatible with express-session's

* docs: update Typescript support section

RafaelGSS committed:
feat: add ignoreFields option to regenerate method (#189)

Signed-off-by: RafaelGSS <rafael.nunu@hotmail.com>

mcollina committed:
Bumped v10.2.0

Signed-off-by: Matteo Collina <hello@matteocollina.com>

ohchi committed:
fix: separate an actual path from url search part before verification (#192)

Eomm committed:
fix: lint (#195)

Eomm committed:
Bumped v10.2.1

climba03003 committed:
feat!: stricter types (#196)

* refactor: move FastifySessionObject inside namespace to allow override

* feat: stricter types

mcollina committed:
Bumped v10.3.0

Signed-off-by: Matteo Collina <hello@matteocollina.com>

Fdawgs committed:
ci: only trigger on pushes to main branches (#198)

dependabot[bot] committed:
build(deps-dev): bump @types/node from 18.16.5 to 20.1.0 (#199)

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 18.16.5 to 20.1.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

SimenB committed:
fix: manually persiting the dession should make `isModified` return false (#201)

mcollina committed:
Bumped v10.3.1

Signed-off-by: Matteo Collina <hello@matteocollina.com>

SimenB committed:
fix: make sure to set cookie on manual session saves (#203)

mcollina committed:
Bumped v10.4.0

Signed-off-by: Matteo Collina <hello@matteocollina.com>

dependabot[bot] committed:
build(deps-dev): bump @fastify/cookie from 8.3.0 to 9.0.4 (#205)

Bumps [@fastify/cookie](https://github.com/fastify/fastify-cookie) from 8.3.0 to 9.0.4.
- [Release notes](https://github.com/fastify/fastify-cookie/releases)
- [Commits](https://github.com/fastify/fastify-cookie/compare/v8.3.0...v9.0.4)

---
updated-dependencies:
- dependency-name: "@fastify/cookie"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

zanechua committed:
chore: update redis example (#207)

dependabot[bot] committed:
build(deps-dev): bump tsd from 0.28.1 to 0.29.0 (#208)

Bumps [tsd](https://github.com/SamVerschueren/tsd) from 0.28.1 to 0.29.0.
- [Release notes](https://github.com/SamVerschueren/tsd/releases)
- [Commits](https://github.com/SamVerschueren/tsd/compare/v0.28.1...v0.29.0)

---
updated-dependencies:
- dependency-name: tsd
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tomoakiichige committed:
fix: remove cookie.expires from shouldSaveSession evaluation logic (#209)

* fix: remove cookie.expires from shouldSaveSession evaluation logic

* test: add test

mcollina committed:
Bumped v10.5.0

Signed-off-by: Matteo Collina <hello@matteocollina.com>

Fdawgs committed:
perf: use `node:` prefix to bypass require.cache call for builtins (#210)

See https://github.com/fastify/fastify-static/pull/407

Fdawgs committed:
chore: add `.gitattributes` file (#212)

Fdawgs committed:
perf(lib/fastifysession): optimize split param (#214)

Fdawgs committed:
chore(package): explicitly declare js module type (#217)

VIEWVIEWVIEW committed:
Added documentation for import of modules w/ declaration merging (#220)

I ran into ```Property 'session' does not exist on type 'FastifyRequest<RouteGenericInterface, RawServerDefault, IncomingMessage, FastifySchema, FastifyTypeProviderDefault, unknown, FastifyBaseLogger, ResolveFastifyRequestType<...>>'.``` for half an hour, because I used ``const fastifySession = require("@fastify/session")`` rather than the imports. Commonjs requires don't work with module declarations (which I didn't know beforehand). This will probably save someone else's time :P

maxt41 committed:
fix: updated import as per redis-connect docs (#219)

gurgunday committed:
export cookie types (#221)

* try to export cookie's type

* avoid deep require

* update

gurgunday committed:
v10.6.0

gurgunday committed:
use CookieSerializeOptions (#223)

gurgunday committed:
v10.6.1

nicob-29 committed:
Fix: add ability to create Partitioned Cookies (#226)

* Partitioned Bis

* Update lib/cookie.js

Co-authored-by: Manuel Spigolon <behemoth89@gmail.com>

* Update lib/cookie.js

Co-authored-by: Gürgün Dayıoğlu <gurgun.dayioglu@icloud.com>

* Update package.json

Co-authored-by: Aras Abbasi <aras.abbasi@googlemail.com>

* Add tests

* Update test/cookie.test.js

Co-authored-by: Aras Abbasi <aras.abbasi@googlemail.com>

* Update README.md

Co-authored-by: Gürgün Dayıoğlu <gurgun.dayioglu@icloud.com>

---------

Co-authored-by: Manuel Spigolon <behemoth89@gmail.com>
Co-authored-by: Gürgün Dayıoğlu <gurgun.dayioglu@icloud.com>
Co-authored-by: Aras Abbasi <aras.abbasi@googlemail.com>

mcollina committed:
Bumped v10.7.0

Signed-off-by: Matteo Collina <hello@matteocollina.com>

dependabot[bot] committed:
build(deps-dev): bump tsd from 0.29.0 to 0.30.1 (#231)

Bumps [tsd](https://github.com/tsdjs/tsd) from 0.29.0 to 0.30.1.
- [Release notes](https://github.com/tsdjs/tsd/releases)
- [Commits](https://github.com/tsdjs/tsd/compare/v0.29.0...v0.30.1)

---
updated-dependencies:
- dependency-name: tsd
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Fdawgs committed:
docs(readme): replace `fastify.io` links with `fastify.dev` (#232)

Commit:
chore(license): Update licensing year (#233)

Fdawgs committed:
chore(.gitignore): add .tap/ dir (#235)

dependabot[bot] committed:
build(deps-dev): bump cronometro from 1.2.0 to 3.0.1 (#236)

Bumps [cronometro](https://github.com/ShogunPanda/cronometro) from 1.2.0 to 3.0.1.
- [Release notes](https://github.com/ShogunPanda/cronometro/releases)
- [Changelog](https://github.com/ShogunPanda/cronometro/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ShogunPanda/cronometro/compare/v1.2.0...v3.0.1)

---
updated-dependencies:
- dependency-name: cronometro
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mohd-akram committed:
fix: use request.protocol to check for HTTPS (#238)

dependabot[bot] committed:
build(deps-dev): bump tsd from 0.30.7 to 0.31.0 (#239)

Bumps [tsd](https://github.com/tsdjs/tsd) from 0.30.7 to 0.31.0.
- [Release notes](https://github.com/tsdjs/tsd/releases)
- [Commits](https://github.com/tsdjs/tsd/compare/v0.30.7...v0.31.0)

---
updated-dependencies:
- dependency-name: tsd
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

autopulated committed:
fix race condition in cookie.maxAge test (#242)

chuanqisun committed:
ensure maxAge type has milliseconds as the unit (#245)

Co-authored-by: Chuanqi <chuanqi.sun.dev@gmail.com>

gurgunday committed:
v10.7.1

Signed-off-by: Gürgün Dayıoğlu <hey@gurgun.day>

autopulated committed:
enhance checkOptions to reject invalid signer objects, and test. This catches accidentally passing a Buffer instead of a string. (#241)

mcollina committed:
Bumped v10.7.2

Signed-off-by: Matteo Collina <hello@matteocollina.com>

mcollina committed:
use sha256 in place of sha1 (#247)

Signed-off-by: Matteo Collina <hello@matteocollina.com>

zetaraku committed:
Migrate from connect-redis v6 (#248)

zetaraku committed:
Fix type argument inference of `session.get(key)` and `session.set(key, value)` (#249)

* Update types.d.ts

Signed-off-by: Raku Zeta <zetaraku@gmail.com>

* Update types.test-d.ts

Signed-off-by: Raku Zeta <zetaraku@gmail.com>

---------

Signed-off-by: Raku Zeta <zetaraku@gmail.com>

mcollina committed:
Bumped v10.8.0

Signed-off-by: Matteo Collina <hello@matteocollina.com>

ShogunPanda committed:
Merge pull request from GHSA-pj27-2xvp-4qxg

* fix: Use original expiration date when evaluating restored session.

* fix: Also serialize new field.

mcollina committed:
Bumped v10.9.0

Signed-off-by: Matteo Collina <hello@matteocollina.com>

oom- committed:
fix(session): parameter not placed at good position in reload with callback

oom- committed:
refactor(session): improve callback/promise handling

mcollina reviewed (changes requested):
Unfortunately these are not equivalent as their error handling would be different.

jsumners committed:
Merge `next` into `master` (#256)

* ci

* update tap

Co-authored-by: Gürgün Dayıoğlu <hey@gurgun.day>

jsumners committed:
v11.0.0-pre.fv5.1

dependabot[bot] committed:
build(deps): bump fastify/workflows from 4.1.0 to 4.2.1 (#258)

Bumps [fastify/workflows](https://github.com/fastify/workflows) from 4.1.0 to 4.2.1.
- [Release notes](https://github.com/fastify/workflows/releases)
- [Commits](https://github.com/fastify/workflows/compare/v4.1.0...v4.2.1)

---
updated-dependencies:
- dependency-name: fastify/workflows
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

zetaraku committed:
improve type test readability (#259)

oom- committed:
Merge branch 'fastify:master' into master
